share/ansible/roles/ci_run/tasks/fedora.yml

```diff
@@ -0,0 +1,69 @@
+---
+# tasks file for ci_run
+- name: Ensure dependencies are installed
+  ansible.builtin.dnf:
+    name:
+      - dnf-plugins-core
+      - libcmocka-devel
+      - systemd-devel
+    state: present
+
+- name: Ensure build dependencies are installed
+  ansible.builtin.command:
+    dnf builddep -y shadow-utils
+  register: dnf_result
+  changed_when: '"Nothing to do" not in dnf_result.stdout'
+
+- name: Build configuration
+  ansible.builtin.command: >
+    ./autogen.sh
+    --disable-account-tools-setuid
+    --enable-lastlog
+    --enable-logind=no
+    --enable-man
+    --enable-shadowgrp
+    --enable-shared
+    --with-audit
+    --with-bcrypt
+    --with-group-name-max-length=32
+    --with-libpam
+    --with-selinux
+    --with-sha-crypt
+    --with-yescrypt
+    --without-libbsd
+    --without-libcrack
+    --without-sssd
+  args:
+    chdir: /usr/local/src/shadow/
+  ignore_errors: true
+
+- name: Build
+  ansible.builtin.shell:
+    make -Orecurse -j4 > build.log
+  args:
+    chdir: /usr/local/src/shadow/
+  ignore_errors: true
+
+- name: Run unit-tests
+  ansible.builtin.command:
+    make check
+  args:
+    chdir: /usr/local/src/shadow/
+  ignore_errors: true
```

Comment:
What does this mean?  Why do we want to ignore errors?

Reply:
By default if an ansible action fails, then the ansible execution is stopped. I want to ignore the errors and continue the execution to run the last action where the logs are copied from the container to the host system. This way we can gather them for inspection.

Reply:
Thanks!

Reply:
BTW, I see in that run that the store artifacts step isn't being run?

Reply:
My bad, I forgot to set [if: always()](https://github.com/shadow-maint/shadow/pull/1009/files#diff-5c8806e0119aa46657122645a786c2d9caaf3d30652775f2ae4dc45ffce209cdR111) in the Github Action. It's fine now.

Reply:
Thanks!  It seems to work now.  BTW, dumb question: how do I find and read the logs (artifacts)?

Reply:
You need to open the [action that failed](https://github.com/shadow-maint/shadow/actions/runs/9385996635/job/25845492826?pr=1009), then click on [summary](https://github.com/shadow-maint/shadow/actions/runs/9385996635?pr=1009), and finally scroll down to find all the artifacts

Reply:
Ughhh, and then download a .zip and extract it to find the logs.  Can we have a copy on stderr?  I very much prefer scrolling.  :)

Reply:
That will work until we start running the new tests and have several files to review